modules/apps/app-builder/app-builder-test/src/testFunctional/tests/AppBuilder.testcase

```diff
@@ -1290,6 +1290,79 @@ definition {
 		AppBuilderAdmin.validateItemNameOnTable(itemName = "123");
 	}
 
+	@description = "Create more than one Table View and choose between the two when deploying an App"
+	@priority = "5"
+	test DeployAppChoosingBetweenDifferentTableViews {
+
+		AppBuilderAdmin.addCustomObject(
+			customObjectName = "Custom Object Title",
+			isCreatingFormView = "true",
+			isCreatingViaPlusIcon = "true");
+
+		AppBuilderAdmin.createFormViewAndTableView(
+			fieldType = "Text",
+			formViewName = "Text Form View",
+			tableViewName = "My Table View 1");
+
+		AppBuilderAdmin.switchTab(tabName = "Form Views");
+
+		AppBuilderAdmin.createFormViewAndTableView(
+			fieldType = "Upload",
+			formViewName = "Upload Form View ",
+			tableViewName = "My Table View 2");
+
+		AppBuilderAdmin.switchTab(tabName = "Apps");
+
+		AppBuilderAdmin.add();
+
+		AppBuilderAdmin.setName(title = "My First App");
+
+		App.selectFormView(formViewName = "Text Form View");
+
+		App.selectTableView(tableViewName = "My Table View 2");
+
+		Button.click(button = "Previous");
+
+		App.selectTableView(tableViewName = "My Table View 1");
+
+		App.deployAppToggleSwitcher(deployOption = "Widget");
+
+		Button.click(button = "Deploy");
+
+		Navigator.openURL();
+
+		Portlet.addPG(portletName = "My First App");
+
+		Button.click(button = "New Entry");
+
+		Button.click(button = "Save");
+
+		Portlet.addPG(portletName = "My First App (Table View)");
+
+		TableViewBuilder.validatePortletFieldType(fieldType = "Text");
+
+		ProductMenu.gotoPortlet(
+			category = "App Builder",
+			panel = "Control Panel",
+			portlet = "Custom Objects");
+
+		AppBuilderAdmin.openItem(itemName = "Custom Object Title");
+
+		AppBuilderAdmin.switchTab(tabName = "Apps");
+
+		AppBuilderAdmin.openItem(itemName = "My First App");
+
+		Button.click(button = "Next");
```

Comment:
Hi Tai, I would suggest you the same suggestion of before, to add a new macro App.macro like:
clickNext {
Button.click(button = "Next");
}
Or use like Broke did in the App.selectFormView or App.selectTableView, where she called Button.clickNext(); directly.